### PR TITLE
Publish results endpoint

### PR DESCRIPTION
**Pull request by leio10**
This PR adds the publish results endpoint, to allow authorities to make election results public.

It also improves the `status` enums for the elections and pending messages at a database level, using integers instead of strings and setting the right default values.

### CODE REVIEW

**Review by beagleknight — APPROVED** [review]
Great job! 🤩

File: decidim-bulletin_board-app/spec/commands/publish_results_spec.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    end
+  end
+
+  context "when the client is not the election authority" do
+    let(:client) { create(:authority, private_key: private_key) }
+    let(:private_key) { generate(:private_key) }
+
+    it_behaves_like "publishing results fails"
+
+    it "broadcast invalid" do
```
Comment (slickepinne) [inline decidim-bulletin_board-app/spec/commands/publish_results_spec.rb:65]:
```suggestion
    it "broadcasts invalid" do
```

File: decidim-bulletin_board-app/spec/commands/publish_results_spec.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    end
+  end
+
+  context "when the client is a trustee" do
+    let(:client) { Trustee.first }
+    let(:private_key) { Test::PrivateKeys.trustees_private_keys.first }
+
+    it_behaves_like "publishing results fails"
+
+    it "broadcast invalid" do
```
Comment (slickepinne) [inline decidim-bulletin_board-app/spec/commands/publish_results_spec.rb:76]:
```suggestion
    it "broadcasts invalid" do
```

File: decidim-bulletin_board-app/spec/commands/publish_results_spec.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      expect { subject }.to broadcast(:invalid, "Invalid client")
+    end
+  end
+
+  context "when the message author is not the authority" do
+    let(:extra_message_params) { { authority: create(:authority) } }
+
+    it_behaves_like "publishing results fails"
+
+    it "broadcast invalid" do
```
Comment (slickepinne) [inline decidim-bulletin_board-app/spec/commands/publish_results_spec.rb:86]:
```suggestion
    it "broadcasts invalid" do
```

File: decidim-bulletin_board-app/spec/mutations/publish_results_mutation_spec.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    let!(:election) { create(:election, status: :results) }
+    let(:authority) { Authority.first }
+    let(:headers) { { "Authorization": authority.api_key } }
+    let(:signed_data) { JWT.encode(payload.as_json, Test::PrivateKeys.authority_private_key.keypair, "RS256") }
+    let(:payload) { build(:publish_results_message, election: election) }
+    let(:message_id) { payload["message_id"] }
+
+    it "changes the election status" do
+      expect { subject }.to change { Election.last.status } .from("results") .to("results_published")
```
Comment (slickepinne) [inline decidim-bulletin_board-app/spec/mutations/publish_results_mutation_spec.rb:31]: Are these spaces on purpose? Before `.from` and `.to`?

File: decidim-bulletin_board-ruby/spec/decidim/bulletin_board/authority/publish_results_spec.rb [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            }
+          }
+        end
+
+        context "when everything is ok" do
+          it "broadcasts ok with the result of the graphql mutation" do
+            expect { subject.call }.to broadcast(:ok)
+          end
+
+          it "uses the graphql client to publish the election results and return its result" do
```
Comment (slickepinne) [inline decidim-bulletin_board-ruby/spec/decidim/bulletin_board/authority/publish_results_spec.rb:30]:
```suggestion
          it "uses the graphql client to publish the election results and returns its result" do
```

**Review by slickepinne — APPROVED** [review]
Nice job! 🎉

File: decidim-bulletin_board-app/spec/mutations/publish_results_mutation_spec.rb (hunk 4 above)
Comment (leio10) [reply decidim-bulletin_board-app/spec/mutations/publish_results_mutation_spec.rb]: Nope, I've changed it 👍

**Pull request merged**